### PR TITLE
Update Doc Argo CD Configuration

### PR DESCRIPTION
**Pull request by clementblaise**
### Description of your changes

Simplify the current configuration and add additional tweaks

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: docs/guides/argo-cd-crossplane.md [hunk 1]
```diff
@@ -7,32 +7,54 @@ weight: 270
...
-### Configuring Argo CD with Crossplane
- 
-To configure Argo CD for Annotation resource tracking, edit the `argocd-cm`
-
-`ConfigMap` in the `argocd` `Namespace`. Add `application.resourceTrackingMethod: annotation`
-
-to the data section as below:
+### Resource Tracking with annotations
 
+In order for Argo CD to properly track Crossplane generated resources, tracking method must be updated to annotations :
```
Comment (negz) [inline docs/guides/argo-cd-crossplane.md:16]: Could you please wrap docs at 80 chars, in line with the existing file. There's a few plugins that will do this.

Comment (jeanduplessis) [reply docs/guides/argo-cd-crossplane.md:16]: @negz have we considered adding a linter to the CI pipeline to check and report on this? If not I can open an issue to look to introduce this.

Comment (negz) [reply docs/guides/argo-cd-crossplane.md]: @jeanduplessis Happy to have a linter handle this, please go ahead.

**Review by turkenh — APPROVED** [review]
Thank you @clementblaise!

**Pull request merged**